drivers/display/display_sh1122.c

```diff
@@ -0,0 +1,515 @@
+
+#define SH1122_DEFINE_MIPI(node_id)                                                                \
+	static uint8_t conversion_buf##node_id[SH1122_CONV_BUFFER_SIZE];                           \
+	static struct sh1122_data data##node_id;                                                   \
+	static const struct sh1122_config config##node_id = {                                      \
+		.mipi_dev = DEVICE_DT_GET(DT_PARENT(node_id)),                                     \
+		.dbi_config = MIPI_DBI_CONFIG_DT(                                                  \
+			node_id, SH1122_WORD_SIZE(node_id) | SPI_OP_MODE_MASTER, 0),               \
+		.height = DT_PROP(node_id, height),                                                \
+		.width = DT_PROP(node_id, width),                                                  \
+		.oscillator_freq = DT_PROP(node_id, oscillator_freq),                              \
+		.display_offset = DT_PROP(node_id, display_offset),                                \
+		.start_line = DT_PROP(node_id, start_line),                                        \
+		.multiplex_ratio = DT_PROP(node_id, multiplex_ratio),                              \
+		.color_inversion = DT_PROP(node_id, inversion_on),                                 \
+		.phase_length = DT_PROP(node_id, phase_length),                                    \
+		.dc_dc = DT_PROP(node_id, dc_dc),                                                  \
+		.precharge_voltage = DT_PROP(node_id, precharge_voltage),                          \
+		.vcomh_voltage = DT_PROP(node_id, vcomh_voltage),                                  \
+		.low_voltage = DT_PROP(node_id, low_voltage),                                      \
+		.inv_seg = DT_PROP(node_id, inv_seg),                                              \
+		.inv_com = DT_PROP(node_id, inv_com),                                              \
+		.write_cmd = sh1122_write_bus_cmd_mipi,                                            \
+		.write_pixels = sh1122_write_pixels_mipi,                                          \
+		.release_bus = sh1122_release_bus_mipi,                                            \
+		.conversion_buf = conversion_buf##node_id,                                         \
+		.conversion_buf_size = sizeof(conversion_buf##node_id),                            \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_DEFINE(node_id, sh1122_init, NULL, &data##node_id, &config##node_id,             \
+			 POST_KERNEL, CONFIG_DISPLAY_INIT_PRIORITY, &sh1122_driver_api);
+
+#define SH1122_DEFINE(node_id)                                                                     \
+	COND_CODE_1(DT_ON_BUS(node_id, i2c), \
+	(SH1122_DEFINE_I2C(node_id)), (SH1122_DEFINE_MIPI(node_id)))
+
+DT_FOREACH_STATUS_OKAY(sinowealth_sh1122, SH1122_DEFINE)
```

Comment:
It seems more frequent for drivers to have a [`DT_INST_FOREACH_STATUS_OKAY()` using `#define DT_DRV_INST`](https://docs.zephyrproject.org/latest/build/dts/howtos.html#option-1-create-devices-using-instance-numbers) and a `n` (instance number) instead of `node_id`.

However, I have no idea why one style would be better than the other! Maybe just habits...